store/src/main/java/com/zimbra/cs/service/admin/IssueCert.java

```diff
@@ -0,0 +1,114 @@
+ * {@link com.zimbra.cs.rmgmt.RemoteManager}, {@link RemoteCertbot}.
+ *
+ * @author Yuliya Aheeva
+ * @since 23.3.0
+ */
+public class IssueCert extends AdminDocumentHandler {
+
+  /**
+   * Handles the request. Searches a domain by id, checks admin rights (accessible to global and
+   * delegated admin of requested domain), searches a server with proxy node, executes remote
+   * certbot command on it, creates response element.
+   *
+   * @param request {@link Element} representation of {@link
+   *     com.zimbra.soap.admin.message.IssueCertRequest}
+   * @param context request context
+   * @return {@link Element} representation of {@link
+   *     com.zimbra.soap.admin.message.IssueCertResponse}
+   * @throws ServiceException in case if domain could not be found, domain doesn't have
+   * PublicServiceHostname and at least one VirtualHostName,
+   * server with proxy node could not be found or domain admin dosn't have rights to deal with
+   * this domain.
+   *
+   * It won't throw an exception in case if remote execution command fails, instead will create
+   * a response and add a failure message to it.
+   */
+  @Override
+  public Element handle(final Element request, final Map<String, Object> context)
+      throws ServiceException {
+
+    ZimbraSoapContext zsc = getZimbraSoapContext(context);
+
+    Provisioning prov = Provisioning.getInstance();
+    String domainId = request.getAttribute(AdminConstants.A_DOMAIN);
+    Domain domain = prov.get(DomainBy.id, domainId);
+
+    if (domain == null) {
+      throw ServiceException.INVALID_REQUEST(
+          "Domain with id " + domainId + " could not be found.", null);
+    }
+
+    AdminAccessControl admin = checkDomainRight(zsc, domain, AdminRights.R_getDomain);
+    String adminMail = admin.mAuthedAcct.getMail();
+
+    String chain = request.getAttribute(AdminConstants.A_CHAIN, AdminConstants.DEFAULT_CHAIN);
+    boolean expand = request.getAttributeBool(MailConstants.A_EXPAND, false);
+
+    String publicServiceHostname = domain.getPublicServiceHostname();
+    String[] virtualHostNames = domain.getVirtualHostname();
+    if (publicServiceHostname == null || virtualHostNames.length == 0) {
+      throw ServiceException.INVALID_REQUEST(
+          "Domain with id " + domainId + " must have PublicServiceHostname and VirtualHostName.",
+          null);
+    }
+
+    // First release will work only on ONE proxy even if there are multiple proxy in the infrastructure.
+    Optional<Server> optionalServer =
+        prov.getAllServers()
+            .stream()
+            .filter(Server::hasProxyService)
+            .findFirst();
+
+    if (optionalServer.isEmpty()) {
+      throw ServiceException.NOT_FOUND("Server with carbonio-proxy node could not be found.");
+    }
+
+    ZimbraLog.rmgmt.info("Issuing a LetsEncrypt cert for domain " + domainId);
+
+    RemoteManager remoteManager = RemoteManager.getRemoteManager(optionalServer.get());
```

Comment:
Another kung-fu java functional style :call_me_hand: 

```suggestion
    final RemoteManager remoteManager =
        prov.getAllServers()
            .stream()
            .filter(Server::hasProxyService)
            .findAny() // just find the first suitable, faster search
            .map(server -> {
                  ZimbraLog.rmgmt.info("Issuing a LetsEncrypt cert for domain " + domainId);
                 return RemoteManager.getRemoteManager(server);
            })
            .orElseThrow(() -> ServiceException.NOT_FOUND("Server with carbonio-proxy node could not be found."))
```

Reply:
i dont wanna do this kung-fu it needs to be surrounded with try catch for remote manager

Reply:
Maybe I'm missing something, but I don't see any try/catch block in the method body :thinking:

Reply:
Fixed irl